packages/mui-styled-engine-sc/src/index.js

```diff
@@ -36,6 +36,10 @@ export default function styled(tag, options) {
   return stylesFactory;
 }
 
+Object.keys(scStyled).forEach((tag) => {
```

Comment:
We will need to update the type of the styled utility too, to extend `ThemedStyledComponentFactories<Theme>`

Reply:
I'm not entirely sure which type needs to be updated, but if you let me know, I can make that change!

Reply:
Similar to how the types for the @mui/styled-engine are updated. The changes would need to be made in packages/mui-styled-engine-sc/src/index.d.ts